src/Control/Eff/QuickStart.hs

```diff
@@ -0,0 +1,226 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# OPTIONS_GHC -Wno-simplifiable-class-constraints #-}
+
+-- | This module contains several tiny examples of how to use effects.
+--
+-- @
+-- import Control.Eff
+-- import Control.Eff.Reader.Lazy
+-- import Control.Eff.Writer.Lazy
+-- import Control.Eff.State.Lazy
+-- import Control.Eff.Exception
+-- @
+
+module Control.Eff.QuickStart where
+
+import           Control.Eff
+import           Control.Eff.Reader.Lazy
+import           Control.Eff.State.Lazy
+import           Control.Eff.Exception
+import           Control.Monad                            ( when )
+
+
+-- | an effectful function that can throw an error
+--
+-- @
+-- tooBig = do
+--   when (i > 100) $ throwError $ show i
+--   return i
+-- @
+tooBig :: Member (Exc String) r => Int -> Eff r Int
+tooBig i = do
+  when (i > 100) $ throwError $ show i
+  return i
+
+-- | run the @tooBig@ effect based on a provided Int.
+--
+-- @
+-- runTooBig i = run . runError $ tooBig i
+-- @
+runTooBig :: Int -> Either String Int
+runTooBig i = run . runError $ tooBig i
+
+-- | run the @tooBig@ effect, giving @1@ as input
+--
+-- @
+-- runTooBig1 = runTooBig 1
+-- -- Right 1
```

Comment:
@power-fungus these should either be moved to `Control.Eff.Quickstart.Test` module. alternatively, you could use something like [doctest](http://hackage.haskell.org/package/doctest)

Reply:
I change to doctest